### PR TITLE
Polidea update - Documentation

### PR DESCRIPTION
**Pull request by Jakub-Sadowski**
PR contains : documentation page and new sidebars styles

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by craigbox** [comment]
@ericvn @frankbu @brian-avery @therealmitchconnors @nmittler @linsun @istio/wg-docs-maintainers @istio/steering-committee @rvennam @christian-posta @ZackButcher @howardjohn 

Our site redesign work has hit the point where it touches the documentation. [Here is the approved mockup](https://xd.adobe.com/view/173cb1e8-e371-4354-bbf5-ef8330df6b90-426f/screen/46b39fc3-c117-429e-b968-f1edf76d6f33).

There will be lots of areas where the actual implementation needs tweaking, that the mockup didn't account for: for example, one that I am seeing immediately is how the sidebar works when it's more than three levels deep.

PTAL at [the preview](https://deploy-preview-9121--istio-polidea.netlify.app/latest/docs/) and hit a few pages you care about.  Please offer up any easily addressable feedback as comments on this thread.  For "I don't think it should be designed that way" or things that you think will require larger effort, please create new issues and assign them to me.

Once these PRs are approved, we'll be looking to merge this branch to preliminary in a couple of weeks.  The Polidea team will roll off at that point and further changes will be on us.

**Comment by howardjohn** [comment]
Initial thoughts, from someone with 0 design experience. I am comparing this to linkerd, knative, gateway-api.org, and kubernetes.

* It is *very* white. I would recommended we make the header not white at least?. The white on the left side of the sidebar also feels a bit awkward as it makes the sidebar look like its floating
* The header (Istio logo + links to about/docs/blog) is really big - far bigger than any similar sites I could find.
* The 3 buttons on the right side (translate, edit, report site bug) are out of place in the new design
* On my phone, I still have dark mode. This is out of place with the new design, and I don't think there is anyway to actually change it in the new design? Ideally we can still switch between dark and light mode, and dark mode is updated as well.
* Its fairly hard to find the "version" selection. All other sites have this in the header
* The view when TOC/sidebar are collapsed wastes over half the screen (will attach picture
* Padding on copy button in code view is off
* I am sure this is highly personal preference, but I find the font hard to read for extended times.
* Sidebar does not scroll but its longer than my screen - it only scrolls if the body is at the bottom of the page as well
* Sidebar is too narrow/has too much padding so almost ever single entry is wrapped
* Table format doesn't seem consistent with the tab format - at least the blue color is different

**Comment by craigbox** [comment]
Thanks John and Brian.

We'll iterate on the design over time, but right now we're implementing the design we have signed off on, and most of the content is "lorem ipsum" and the things you called out are just placeholders for pages not yet written. Please call out anything you see as rendering incorrectly in the **docs** explicitly; we'll talk more later about general design changes.

A few specifics:

> logo colors

Logos are in color when moused over, and monochrome until then.  We'll check in with the vendors on this pattern.

> doc test indicator

Yes, that's due to come back.

> select older versions of docs

It's in the footer.  I'd be interested in data on how many people need to access it to make it more prominent again; the simplification is through removing things.  Like dark mode.  Which the community might bring back but we excluded from this design. :)

**Comment by brian-avery** [comment]
Thoughts on the versions and I'll dig into the docs:

* we know a very large number of Istio users are still using old versions. Last we checked, many of these are on 1.5. These users may still find the need to be able to find the docs. 
* I don't see the security announcements listed anywhere.

**Comment by brian-avery** [comment]
On Firefox 85 on Fedora, it renders as below. Alignment of the search icon is off, but coloring is consistent.

image

**Comment by ericvn** [comment]
First thing I noticed (Safari on the Mac) was what Brian showed with the page being mostly non-white. I initially didn't understand John's comment about too much white until I looked in the design and saw those being white. I initially figured that maybe white wasn't going to be used and had a todo to ask which way was correct.  Sounds like something needs to be fixed on the Mac.

**Comment by aijamalnk** [comment]
Hi all!
I'm Aizhamal, and I've been working with the team managing the project for Google. To help our vendors, I will try to condense the comments on functionality that should be fixed for this PR.

Should be fixed:

- The dark blue banner does not show up for me in Chrome ([image](https://user-images.githubusercontent.com/14306210/110732471-14b19200-81d9-11eb-8410-6d0c6101f837.png)), but in Safari it does show up. **I think this should be made consistent.**
- I also experienced the blue banner blocking the menu ([image](https://user-images.githubusercontent.com/14306210/110732778-a6b99a80-81d9-11eb-9130-0a3867decbb6.png)). **Can we have this fixed please?**
- We should also **make sure that the search icon is properly aligned** (mentioned in [this comment](https://github.com/istio/istio.io/pull/9121#issuecomment-795563522))
    - I noticed other odd things when the window is small in Safari([image](https://user-images.githubusercontent.com/14306210/110733334-92c26880-81da-11eb-9cfe-5b272b5cdcc8.png)). I was unable to scroll up to use the search bar, we also cannot close the menu. This is bad for mobile devices, where users need to refresh or go somewhere on the menu if they want to close it. **Let's make sure the menu can be closed, and the search bar is accessible.**
    - the menu took over the whole page, and half of the screen was just white, which is also confusing. (see image above). **Let's make the menu not cover the whole screen unless necessary**.
- I was able to reproduce the half-screen-white from [previous comments](https://github.com/istio/istio.io/pull/9121#issuecomment-795567568) - but only once - I was not able to make it happen again (neither Safari nor Chrome). It would be great to fix this, but if we can't reproduce, then we will have to leave it for later.
- As dark mode is out-of-scope for this project, we should probably disable it, so that users won't see the half-baked view from [previous comments](https://github.com/istio/istio.io/pull/9121#issuecomment-794767923) - @Jakub-Sadowski and team promised to try to make it work like 'normal' by adding the same "schema", but can't dedicate much time to it.

Issues where there are questions but are not binding:
- The padding for copy button changed. Though I thought it would instead show up in the corner - it is showing up outside of the code box ([image](https://user-images.githubusercontent.com/14306210/110732238-a371df00-81d8-11eb-9411-850457aa7d68.png)) - is this the way other sites do it?

Issues that have been addressed (thank you!)
- Side bar padding looks much better now without the extra padding
- Side bar scrolling works better now.
- Jakub mentioned that information about tests in docs has been added back.

**Comment by brian-avery** [comment]
Thanks @aijamalnk. It looks like some of the CSS changes took care of the responsiveness issues. Resizing the windows now seems to work. 

Did the site redesign completely get rid of security releases, upgrade notes, and release notes or is that all for future design issues? If I understand correctly, the design is done and the next step is to merge this into preliminary.istio.io or is this iterative?

**Comment by brian-avery** [comment]
The copy button doesn't always seem to align with text boxes. See

image
on https://deploy-preview-9121--istio-polidea.netlify.app/latest/docs/tasks/traffic-management/tcp-traffic-shifting/

**Comment by brian-avery** [comment]
Along the lines of responsiveness, not sure if this is an issue or not, but there's still *a lot* of whitespace on larger screens

image

File: assets/sass/misc/_banners.scss
Comment (howardjohn) [inline assets/sass/misc/_banners.scss:3]: Can we change the headerHeight? It is the largest header on any site I use, by almost a 2x margin. It takes up nearly 1/5 of my screen

**Comment by craigbox** [comment]
Brian, re. whitespace: this is somewhat deliberate.  Look at this very GitHub screen for example.

image

#meta

I had the opposite complaint about the previous site on my 5K monitor.  I had to constantly move my head left to right to read.  It should have a columnar layout with [about 75 characters per line](https://baymard.com/blog/line-length-readability).  That said, it's currently capped at 1280px, and I'm suggesting we increase it to a max of 1440px (the width of a 13" MBP)

**Comment by brian-avery** [comment]
Thanks @craigbox I've got an ultrawide and wasn't sure if it was an issue but wanted to present it. I usually split the screen so it doesn't matter.

**Comment by howardjohn** [comment]
Table seems inconsistent with the tab and code blobs, should it be more in line with the new style?

**Comment by aijamalnk** [comment]
Issues that have been fixed by @Jakub-Sadowski (thank you!):

1. Blue banner works properly for my browsers. And it looks really nice! Thank you
2. Menu on mobile issues have been fixed: Scrolling and covering-whole-screen are no longer an issue.
3. The half-sreen-white sytuation also fixed 
4. Copy button was modified to always be in top right corner

Can we leave the two following comments for a later PR?
1. Table style is inconsistent with tab style
2. Decrease header height.

LGTM. @davidhauck for comments on code

File: layouts/partials/navigation_level.html
Comment (davidhauck) [inline layouts/partials/navigation_level.html:1]: Is this supposed to be a rename of sidebar_level?

**Comment by aijamalnk** [comment]
> Table seems inconsistent with the tab and code blobs, should it be more in line with the new style?

@howardjohn do you mean that tab and table have different shades of blue, right? or that the Table header and Tab header are not the same blue?

**Comment by howardjohn** [comment]
> > Table seems inconsistent with the tab and code blobs, should it be more in line with the new style?
> 
> @howardjohn do you mean that tab and table have different shades of blue, right? or that the Table header and Tab header are not the same blue?

At least the different shades of blue, but they are also a bit different style (pure blue vs white with blue top highlight). I wouldn't look to me for design advice, I just thought it felt a little out of place, but not a big deal

**Comment by aijamalnk** [comment]
> > > Table seems inconsistent with the tab and code blobs, should it be more in line with the new style?
> > 
> > 
> > @howardjohn do you mean that tab and table have different shades of blue, right? or that the Table header and Tab header are not the same blue?
> 
> At least the different shades of blue, but they are also a bit different style (pure blue vs white with blue top highlight). I wouldn't look to me for design advice, I just thought it felt a little out of place, but not a big deal

thank you for clarifying @howardjohn . I will try to get them in the same style if possible.

File: assets/sass/misc/_banners.scss
Comment (aijamalnk) [reply assets/sass/misc/_banners.scss:3]: @howardjohn do you want to make all of the headers throughout the website use smaller font?

File: layouts/partials/navigation_level.html
Comment (Jakub-Sadowski) [reply layouts/partials/navigation_level.html:1]: It's modified version of sidebar_level made accordingly with new designs, I didn't delete sidebar_level, because I left in primary_top a way to use old sidebars.

File: assets/sass/misc/_banners.scss
Comment (craigbox) [reply assets/sass/misc/_banners.scss:3]: I think he means the total height of the header on the page.

image

It's currently the same height as the main page, when the header is blue.  I've seen other sites that shrink it after it has scrolled; we could go for that but it's not in the design.

Alternatively we could change the headerHeight throughout; if we did that I would recommend we also shrink the boat a little bit, and maybe the wordmark to match.

John, can you confirm yours looks roughly like that (or send a screenshot)?  It's ~13% of the page height for me on my MBP screen, or 1/7.5.  Your screen may be a different aspect ratio.

**Comment by craigbox** [comment]
List pages haven't been formatted to the new design.

https://deploy-preview-9121--istio-polidea.netlify.app/latest/docs/reference/config/networking/

image

File: assets/sass/misc/_banners.scss
Comment (howardjohn) [reply assets/sass/misc/_banners.scss:3]: Yes, that is what mine looks like. I browse on a lot of aspect ratios as well though, and 110-125% zoom

**Comment by Jakub-Sadowski** [comment]
> List pages haven't been formatted to the new design.
> 
> https://deploy-preview-9121--istio-polidea.netlify.app/latest/docs/reference/config/networking/
> 
> image

Sorry but could you send some reference to the design to this page? I can't find appropriate one

**Comment by aijamalnk** [comment]
> > List pages haven't been formatted to the new design.
> > https://deploy-preview-9121--istio-polidea.netlify.app/latest/docs/reference/config/networking/
> > image
> 
> Sorry but could you send some reference to the design to this page? I can't find appropriate one

@Jakub-Sadowski We have minimal changes to the docs in terms of design. We wanted the docs to match the overall design, but wouldn't change anything other than that because we expect the docs remain as it is.

As for your comments: there wasn't a design for docs list items, and what [I see now looks good](https://deploy-preview-9121--istio-polidea.netlify.app/latest/docs/setup/platform-setup/). 

One small request for @Jakub Sadowski
Decrease [this gap](https://user-images.githubusercontent.com/14306210/111368237-d5f55f00-8652-11eb-8efa-49830dfc6e0d.png) to match [the content page gap](https://user-images.githubusercontent.com/14306210/111368270-e4437b00-8652-11eb-8835-c03450f801c9.png)

**Comment by aijamalnk** [comment]
> > > > Table seems inconsistent with the tab and code blobs, should it be more in line with the new style?
> > > 
> > > 
> > > @howardjohn do you mean that tab and table have different shades of blue, right? or that the Table header and Tab header are not the same blue?
> > 
> > 
> > At least the different shades of blue, but they are also a bit different style (pure blue vs white with blue top highlight). I wouldn't look to me for design advice, I just thought it felt a little out of place, but not a big deal
> 
> thank you for clarifying @howardjohn . I will try to get them in the same style if possible.

@howardjohn - we have updated the colors of the tables to match the color of tab. Here is the [screenshot](https://user-images.githubusercontent.com/14306210/111369382-3d5fde80-8654-11eb-8360-80e45d12bf3a.png). Looks much better now. Thanks for your input!

**Comment by howardjohn** [comment]
When we open a page, the sidebar currently retains the state and is expanded to the current page we are on: 


This no longer happens with the new design:

**Comment by howardjohn** [comment]
There is a large empty gap between the content and the header.

In general, it would be good if the content was more prominent proportion of the screen.

This is the current design:


This is how it looks with some tweaks to make the content the focus:


One thing to note with the tweaked design has the exact amount of content as the current istio.io. The current proposed design fits only 106 characters per line, just ~66% of the current istio.io which fits 156 characters.

**Comment by Jakub-Sadowski** [comment]
> I see what you're trying to do, but I don't think style should be inline in the TS; can we keep it all in the CSS please? This will almost certainly bite someone trying to change this in future.

I can change it to adding css class or revert it to the state it was before if you want

**Comment by howardjohn** [comment]
https://deploy-preview-9121--istio-polidea.netlify.app/latest/img/service-mesh.svg is new AFAIK and its not accurate. Metrics are not sent to the control plane. Not sure if that is in scope for this PR though

**Comment by howardjohn** [comment]
I think with the recent padding changes it impacted when the drop shadow comes up for the header.

Scroll down a bit and you see it:

But only scroll down a few ticks and there is no drop shadow:

**Comment by howardjohn** [comment]
The TOC collides with the content here: https://deploy-preview-9121--istio-polidea.netlify.app/latest/docs/reference/config/istio.mesh.v1alpha1/

**Comment by davidhauck** [comment]
lgtm from a code perpective

**Review by davidhauck — APPROVED** [review]

**Comment by aijamalnk** [comment]
> 
> 
> The TOC collides with the content here: https://deploy-preview-9121--istio-polidea.netlify.app/latest/docs/reference/config/istio.mesh.v1alpha1/

Thank you, will address this.

**Pull request merged**
